src/mina-signer/tests/rosetta.test.ts

```diff
@@ -23,6 +51,41 @@ describe('Rosetta', () => {
     client = new Client({ network: 'mainnet' });
   });
 
+  it('field <-> hex roundtrip', () => {
+    const field = BigInt(mainnetSignatureJson.field);
+    const hex = fieldToHex(Field, field);
+    const field_ = fieldFromHex(Field, hex)[0];
+    expect(field_).toEqual(field);
+  });
+
+  it('generates a valid hex string from a signature', () => {
+    let signatureHex = signatureJsonToHex(mainnetSignatureJson);
+    expect(signatureHex).toBe(mainnetSignatureHex);
+  });
+
+  it('generates a valid signature from a hex string', () => {
+    let signature = signatureJsonFromHex(mainnetSignatureHex);
+    expect(signature).toEqual(mainnetSignatureJson);
+  });
+
+  it('signs and verifies signature', () => {
+    let signedTransaction = client.signRosettaTransaction(rosettaUnsignedTxn, privateKey);
+    expect(client.verifyRosettaTransaction(signedTransaction)).toBeTruthy();
+  });
+
+  it('match signature', () => {
+    const { signature } = signTransaction(rosettaUnsignedTxn, privateKey, 'mainnet');
+    expect(signature).toEqual(mainnetSignatureHex);
+  });
+
+  it('verify transaction', () => {
+    expect(client.verifyRosettaTransaction({ data: rosettaUnsignedTxn, signature: mainnetSignatureHex, publicKey: publicKey })).toBeTruthy();
+  });
```

Comment:
maybe add a negative test here? for example, that a client with `network:  'testnet'` returns `false` from `verifyRosettaTransaction`?